### PR TITLE
logging: move backends to subdirectory

### PR DESCRIPTION
**Pull request by cfriedt**
This is mainly to reduce clutter in `subsys/logging`, but also to make backend management slightly easier.

### CODE REVIEW

**Comment by cfriedt** [comment]
* ordered `backends/CMakelists.txt` alphabetically too

**Comment by cfriedt** [comment]
* explicitly add `Kconfig.swo`
* fix compliance issues

**Review by nashif — DISMISSED** [review]
makes sense.

File: subsys/logging/backends/Kconfig.adsp_hda [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	bool "Log backend should pad the buffer with \0 characters when flushing"
+	help
+		HDA requires transfers be 128 byte aligned such that a partial message may
+		never make it across unless padded with \0 characters to the next 128 byte
+		aligned address. This may or may not work depending on the log format
+		being used but should certainly work with text based logging.
+
+endif # LOG_BACKEND_ADSP_HDA
+
+config LOG_BACKEND_ADSP_MTRACE
```
Comment (teburd) [inline subsys/logging/backends/Kconfig.adsp_hda:54]: Kconfig.adsp_mtrace

**Review by teburd — DISMISSED** [review]
Looks good but missed the mtrace Kconfig file split out

File: subsys/logging/backends/Kconfig.adsp_hda (hunk 1 above)
Comment (cfriedt) [reply subsys/logging/backends/Kconfig.adsp_hda]: Fixed!

**Comment by cfriedt** [comment]
* created `Kconfig.adsp_mtrace`

**Review by teburd — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Review by dkalowsk — APPROVED** [review]

**Review by nordic-krch — APPROVED** [review]
Thanks, looks good.

**Review by dcpleung — APPROVED** [review]

**Pull request merged**